cv32e40s/tests/programs/custom/xsecure_test/xsecure_test.c

```diff
@@ -0,0 +1,185 @@
+#ifndef __XSECURE_TEST_H__
+#define __XSECURE_TEST_H__
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+//TODO: functions are fetched from the privilege test code. If these functions end up in a common function file, we should adopt this file to use the common functions.
+
+#define CPUADDR_CPUCTRL     0xBF0
+#define CPUADDR_SECURESEED0 0xBF9
+#define CPUADDR_SECURESEED1 0xBFA
+#define CPUADDR_SECURESEED2 0xBFC
+
+// macros for using defines in inline asm
+#define S(x) #x
+#define STR(s) S(s)
+
+// global values to hold registers for tests
+volatile uint32_t cpuctrl, secureseed0, secureseed1, secureseed2;
+
+int read_ctrlcpu_and_secureseeds_in_user_mode(void);
+int read_ctrlcpu_and_secureseeds_in_machine_mode(void);
+
+
+// assembly function to setup a generous PMP-region for user mode.
+extern volatile void  setup_pmp();
+// assembly function to set privilege-mode to user-mode
+extern volatile void set_u_mode();
+
+// standard value for the mstatus register
+#define MSTATUS_STD_VAL 0x1800
+
+typedef enum {
+  M_MODE_BEH,
+  TRAP_INCR_BEH
+} trap_behavior_t;
```

Comment:
Could we add something like `UNEXPECTED_ERROR_BEH` and let it be set at the end of the handler function every time the handler is invoked?

Reply:
As far as i understand this is almost the same as adding a default case, which I believe will not work that well.